### ISSUE TITLE
systemd (v247) crashes when Type=dbus unit is missing a BusName= attribute

### ISSUE THREAD

**Issue by andir** [systemd/systemd#19920]
**systemd version the issue has been seen with**
 > v247

**Used distribution**
 > NixOS

**Linux kernel version used** (`uname -a`)

 > `5.10.43`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 The unit should be ignored & the user warned as it has an invalid definition.

**Unexpected behaviour you saw**
 systemd crashes during boot when the unit is pulled into multi-user.target

**Steps to reproduce the problem**
 1. Create a new unit of `Type=dbus`, make sure to not include any `BusName=` attribute
 2. Pull the unit into the multi-user.target
 3. Reboot
 4. See systemd crash once it loads the unit.

**Additional program output to the terminal or log subsystem illustrating the issue**

Here is what it looked like on one of my systems: 
![image](https://user-images.githubusercontent.com/638836/121904480-8560d100-cd29-11eb-8913-dc1eb74aadd9.png)

Here is what it looks like when running a (NixOS) VM with a unit as defined above:

```text
machine # <<< NixOS Stage 2 >>>
machine #
machine # [    1.143637] EXT4-fs (vda): re-mounted. Opts: (null)
machine # [    1.144480] booting system configuration /nix/store/9i6gaiyl8fm9rsylabwbc3rpafzbs4jq-nixos-system-machine-21.11pre-git
machine # running activation script...
machine # setting up /etc...
machine # starting systemd...
machine # [    2.088761] systemd[1]: Inserted module 'autofs4'
machine # [    2.108056] systemd[1]: systemd 247 running in system mode. (+PAM +AUDIT -SELINUX +IMA +APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT -GNUTLS +ACL +XZ +LZ4 -ZSTD +SECCOMP +BLKID -ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=unified)
machine # [    2.110166] systemd[1]: Detected virtualization kvm.
machine # [    2.110659] systemd[1]: Detected architecture x86-64.
machine # [    2.111674] systemd[1]: Set hostname to <machine>.
machine # [    2.113483] systemd[1]: Initializing machine ID from random generator.
machine # [    2.158556] systemd-fstab-generator[444]: Checking was requested for "store", but it is not a device.
machine # [    2.162156] systemd-fstab-generator[444]: Checking was requested for "shared", but it is not a device.
machine # [    2.163653] systemd-fstab-generator[444]: Checking was requested for "xchg", but it is not a device.
machine # [    2.389087] systemd[1]: Assertion 'name' failed at src/core/unit.c:3496, function unit_watch_bus_name(). Aborting.
machine # [    2.390202] Core dump to |/bin/false pipe failed
machine # [    2.390716] systemd[1]: Caught <ABRT>, core dump failed (child 452, code=killed, status=6/ABRT).
machine # [    2.391563] systemd[1]: Freezing execution.
```

Unit file equivalent that I used:
```
[Unit]

[Service]

ExecStart=/some/shell/script/that/just/sleeps.sh
Type=dbus

[Install]
WantedBy=multi-user.target
```

**Comment by andir** [comment]
Another way to reproduce this is `systemd-run --unit=dbus1.service --wait -p Type=dbus echo 1` (be careful as your pid1 might segfault if executed as root).

**Comment by yuwata** [comment]
Fix is waiting in #19921.
